bundles/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/config/KNXGenericBindingProvider.java

```diff
@@ -380,10 +386,18 @@ protected KNXBindingConfig parseBindingConfigString(Item item, String bindingCon
                     throw new BindingConfigParseException("Only one readable GA allowed.");
                 }
 
-                Class<? extends Type> typeClass = item.getAcceptedCommandTypes().size() > 0
-                        ? item.getAcceptedCommandTypes().get(i)
-                        : item.getAcceptedDataTypes().size() > 1 ? item.getAcceptedDataTypes().get(i)
-                                : item.getAcceptedDataTypes().get(0);
+                Class<? extends Type> typeClass = null;
+                if( item.getAcceptedCommandTypes().size() > i ) {
+                  typeClass = item.getAcceptedCommandTypes().get(i);
+                }
+                else {
+                  if( item.getAcceptedDataTypes().size() > i ) {
+                    typeClass = item.getAcceptedDataTypes().get(i);
+                  }
+                  else {
+                    typeClass = item.getAcceptedDataTypes().get(0);
+                  }
+                }
```

Comment:
>  wrong that the KNX binding would expect to have knowledge of which States and Commands exist at certain indices in the lists?

After finally reading the wiki, I see that the binding authors direct users to the source code to determine which Types exist at which indices and to write their binding config strings to match.  Maybe not the friendliest approach, but it obviously works. :smile:   

Regarding your fix for index out of bounds, maybe the existing implementation is correct, except that the potential `ArrayIndexOutOfBoundsException` should be caught and a `BindingConfigParseException` should be thrown that explains the mistake in the binding config string that led to the bad list index?  I worry that your fix for the indexing might actually alter how the typeClasses are mapped.